pkg/library/projects/clone.go

```diff
@@ -41,12 +42,54 @@ type Options struct {
 	Env        []corev1.EnvVar
 }
 
+func ValidateAllProjects(workspace *dw.DevWorkspaceTemplateSpec) error {
+	// Temporary additional validation: a project cannot share a name with a dependentProject, and clonePaths cannot collide
+	// Map of project names to project sources (project, dependentProject, starterProject)
+	projectNames := map[string][]string{}
+	// Map of project clone paths to project sources ("project <name>", "starterProject <name>", "dependentProject <name>")
+	clonePaths := map[string][]string{}
+
+	for idx, project := range workspace.Projects {
+		projectNames[project.Name] = append(projectNames[project.Name], "projects")
+		clonePath := GetClonePath(&workspace.Projects[idx])
+		clonePaths[clonePath] = append(clonePaths[clonePath], fmt.Sprintf("project %s", project.Name))
+	}
+
+	for idx, project := range workspace.DependentProjects {
+		projectNames[project.Name] = append(projectNames[project.Name], "dependentProjects")
+		clonePath := GetClonePath(&workspace.Projects[idx])
```

Comment:
The use of `idx` here is because we're taking the reference of the project. Some linters will complain about implicit memory aliasing here (since `&project` is the same address every loop iteration). See e.g. the discussion [here](https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop).

Doesn't impact us here since GetClonePath is just reading fields, but I'd still rather avoid a future false positive.

Reply:
Ahhh makes sense; thank you!

Reply:
No problem :) 

I saw you force pushed some more commits, but [this line](https://github.com/devfile/devworkspace-operator/pull/1205/commits/5299844b5251c363055c4de447b6b107fb185f8a#diff-00dacd00140a354a089d0ae181ad87d399dc92724ca2603dbb0209c5b649c93bR61) still seems to use `&workspace.Projects[idx]` instead of `&workspace.DependentProjects[idx]`